File: between-two-sets/between-two-sets.go
package betweentwosets

func getTotalX(leftNs []int32, rightNs []int32) int32 {
	nStart := leftNs[len(leftNs)-1]
	nEnd := rightNs[0]

	var count int32 = 0

	for n := nStart; n <= nEnd; n += nStart {
		evenlyDivisibleByAllLeftNs := true
		for _, ln := range leftNs {
			if n%ln != 0 {
				evenlyDivisibleByAllLeftNs = false
				break
			}
		}

		if !evenlyDivisibleByAllLeftNs {
			continue
		}

		factorForAllRightNs := true
		for _, rn := range rightNs {
			if rn%n != 0 {
				factorForAllRightNs = false
				break
			}
		}

		if factorForAllRightNs {
			count++
		}
	}

	return count
}

File: between-two-sets/BetweenTwoSets.java
import java.util.List;

public class BetweenTwoSets {

  public static int getTotalX(List<Integer> leftNs, List<Integer> rightNs) {
    int nStart = leftNs.get(leftNs.size() - 1);
    int nEnd = rightNs.get(0);

    int result = 0;

    for (int n = nStart; n <= nEnd; n += nStart) {
      final int nFinal = n;
      
      boolean evenlyDivisibleByAllLeftNs = leftNs
        .stream()
        .allMatch(leftN -> nFinal % leftN == 0);
      
      if (!evenlyDivisibleByAllLeftNs) {
        continue;
      }

      boolean factorForAllRightNs = rightNs
        .stream()
        .allMatch(rightN -> rightN % nFinal == 0);

      if (factorForAllRightNs) {
        result++;
      }
    }

    return result;
  }

}
